src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
+        /// <returns></returns>
+       private Process SetUpProcess(string dll, string level)
+        {
+            Process process = new Process();
+
+            process.StartInfo.FileName = "NUnitTestExtractor.exe";
+            process.StartInfo.Arguments = "-l " + level + " " + dll;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+
+            process.Start();
+
+            return process;
+        }
+
+        [SetUp]
+        public void CreateDirectoryAndFile()
+        {
+            int index = outputDirectory.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
+
+            string directory = "";
+
+            if (index > 0)
+            {
+                directory = outputDirectory.Substring(0, index);
+            }
+
+            if (directory != null && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            if (!File.Exists(outputDirectory))
+            {
+                using (FileStream stream = File.Create(outputDirectory)) { }
+            }
+        } 
+
+        [TearDown]
+        public void EraseOutputFile()
+        {
+            if (File.Exists(outputDirectory))
+            {
+                File.Delete(outputDirectory);
+            }
+        }
+
+        [TestCaseSource(typeof(InvalidDataTestCases))]
+        public void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Empty);
+            }
+        }
+
+        [TestCaseSource(typeof(ValidDataTestCases))]
+        public void WritingToFile_ValidDllAndEveryLevel_GetsWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Not.Empty);
+            }
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Namespace)]
+       public void WritingToFile_LevelIsNamespace_OnlyNamespaceWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, level);
+
+            string[] lines = File.ReadAllLines(outputDirectory);
+            string result = lines[0].Substring(lines[0].IndexOf("|")+2);
```

Comment:
Another way of doing it is using the Split() function instead of Substring().  Something like:
```
string[] parts = lines[0].Split("|");
string testName = parts[1].Trim();
```

or:  `string testName = lines[0].Split("|")[1].Trim();`

Since you do that in multiple tests, you might turn it into private functions like GetDllFromLine() or GetTestNameFromLine().  That way if the line format changes in the future, you'd only need to update one function.